test/integration/Client/client_lib_multi_storytellers.cpp

```diff
@@ -16,80 +19,129 @@ struct thread_arg
 
 chronolog::Client*client;
 
+std::vector <int> shared_state;
```

Comment:
These shared global variables, definitions, and functions are better to be put in a header file and included in each test case.